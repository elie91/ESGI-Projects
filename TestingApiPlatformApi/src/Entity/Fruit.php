<?php

namespace App\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * This is a dummy entity. Remove it!
 * @UniqueEntity(fields={"name"}, message="fruit name need to be unique")
 * @ApiResource(
 *     attributes={
 *          "normalizationContext"={"groups"={"fruit_read", "cart_read", "cart_fruit_read"}},
 *          "denormalizationContext"={"groups"={"fruit_write"}}
 *     }
 * )
 * @ORM\Entity
 *
 */
class Fruit
{
    /**
     * @var int The entity Id
     *
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups({"cart_read"})
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(type="string", length=50)
     * @Assert\Length(
     *      min = 2,
     *      max = 50,
     *      minMessage = "Le nom du fruit doit faire au minimum {{ limit }} caractères",
     *      maxMessage = "Le nom du fruit doit faire au maximum {{ limit }} caractères",
     *      allowEmptyString = false
     * )
     * @Groups({"cart_read"})
     */
    public $name;

    /**
     * @ORM\OneToMany(targetEntity=CartFruit::class, mappedBy="fruit", cascade={"remove"})
     */
    private $cartFruits;

    public function __construct()
    {
        $this->cartFruits = new ArrayCollection();
    }

    /**
     * @return int
     */
    public function getId(): int
    {
        return $this->id;
    }

    /**
     * @param int $id
     */
    public function setId(int $id)
    {
        $this->id = $id;
    }

    /**
     * @return string
     */
    public function getName(): string
    {
        return $this->name;
    }

    /**
     * @param string $name
     */
    public function setName(string $name)
    {
        $this->name = $name;
    }

    /**
     * @return Collection|CartFruit[]
     */
    public function getCartFruits(): Collection
    {
        return $this->cartFruits;
    }

    public function addCartFruit(CartFruit $cartFruit): self
    {
        if (!$this->cartFruits->contains($cartFruit)) {
            $this->cartFruits[] = $cartFruit;
            $cartFruit->setFruit($this);
        }

        return $this;
    }

    public function removeCartFruit(CartFruit $cartFruit): self
    {
        if ($this->cartFruits->contains($cartFruit)) {
            $this->cartFruits->removeElement($cartFruit);
            // set the owning side to null (unless already changed)
            if ($cartFruit->getFruit() === $this) {
                $cartFruit->setFruit(null);
            }
        }

        return $this;
    }
}
