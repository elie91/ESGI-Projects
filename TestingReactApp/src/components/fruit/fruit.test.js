import React from 'react';
import {shallow} from 'enzyme';
import Fruit from './fruit.component';

describe('Fruit component', () => {
  let wrapper;
  let mockAddToCart;
  let mockhHandleDelete;
  const fruit = {id: 1, name: 'poire'};
  const mockItems = [];

  beforeEach(() => {
    mockAddToCart = jest.fn();
    mockhHandleDelete = jest.fn();

    const mockProps = {
      fruit,
      cartItems: mockItems,
      addToCart: mockAddToCart,
      handleDelete: mockhHandleDelete
    };

    wrapper = shallow(<Fruit {...mockProps} />);
  });

  it('should render Fruit component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fruit name as a prop ', () => {
    expect(wrapper.find('.fruit-title').text()).toBe(fruit.name);
  });

  it('should call addToCart when button clicked', () => {
    wrapper.find('.btn-add').simulate('click');
    expect(mockAddToCart).toHaveBeenCalled();
  });

  it('should call handleDelete when button clicked', () => {
    wrapper.find('.btn-delete').simulate('click');
    expect(mockhHandleDelete).toHaveBeenCalled();
  });
});
