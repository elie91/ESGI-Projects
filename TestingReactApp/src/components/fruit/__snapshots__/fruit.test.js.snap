// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fruit component should render Fruit component 1`] = `ShallowWrapper {}`;
